app/code/Magento/InventoryCatalog/Plugin/CatalogInventory/Model/ResourceModel/Stock/Status/AdaptAddStockDataToCollectionPlugin.php

```diff
@@ -55,7 +64,11 @@ public function aroundAddStockDataToCollection(
         $isFilterInStock
     ) {
         $stockId = $this->getStockIdForCurrentWebsite->execute();
-        $this->addStockDataToCollection->execute($collection, (bool)$isFilterInStock, $stockId);
+        if ($stockId === $this->defaultStockProvider->getId()) {
+            $proceed($collection, $isFilterInStock);
```

Comment:
We need to revert this code
We create MySql view for avoiding to have this conditions in code